### ISSUE TITLE
My test dont works in new truffle version (5.0.19)

### ISSUE THREAD

**Issue by rotcivegaf** [trufflesuite/truffle#2046]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

I update the version of truffle(5.0.19), i dont have a migration files, only create the folder in the older version of truffle(5.0.12) i dont have error but now i have an error

How i can use the last vercion of truffle without migrations files?

## Steps to Reproduce

``git clone https://github.com/ripio/rcn-network ``
Open terminal 
``cd rcn-network``
``mkdir migrations``
``npm install``
``npm install -g truffle``
In one terminal ``node_modules/.bin/ganache-cli --gasLimit 7000000``
In other ternminal ``truffle test``

## Expected Behavior

Should pass all tests

## Actual Results

```
> Artifacts written to /tmp/test-119423-28963-vs1tuq.hiz8
> Compiled successfully using:
   - solc: 0.5.8+commit.23d335f2.Emscripten.clang

Error: TypeError: Cannot read property 'filter' of undefined
    at Object.run (/home/------/.nvm/versions/node/v11.5.0/lib/node_modules/truffle/build/webpack:/packages/truffle-migrate/index.js:84:1)
Truffle v5.0.19 (core: 5.0.19)
Node v11.5.0
```

## Environment

* Operating System: linux
* Ethereum client: Ganache CLI v6.4.3 (ganache-core: 2.5.5)
- Truffle version (`truffle version`): 

Old version:
```
Truffle v5.0.12 (core: 5.0.12)
Solidity - 0.5.8 (solc-js)
Node v11.5.0
Web3.js v1.0.0-beta.37
```

New version:

```
Truffle v5.0.19 (core: 5.0.19)
Solidity - 0.5.8 (solc-js)
Node v11.5.0
Web3.js v1.0.0-beta.37
```

* npm version (`npm --version`): 6.4.1

**Comment by gnidan** [comment]
Hey @rotcivegaf thanks for raising this issue. Did you try it on Truffle v5.0.18? Would like to know what the earliest version is where this is broken for you, so that we can look at the changelog for the root cause.

Regardless, we'll look into this issue and see if we can't get it fixed in a jiffy.

**Comment by rotcivegaf** [comment]
This is the latest version that works
```
Truffle v5.0.13 (core: 5.0.13)
Solidity - 0.5.8 (solc-js)
Node v11.5.0
Web3.js v1.0.0-beta.37
```

**Comment by eggplantzzz** [comment]
@rotcivegaf I actually get a different error when I try to follow the steps above with both `truffle@5.0.13` and `truffle@5.0.19`.

`Error: Error: ENOENT: no such file or directory, stat '/Users/tyler/projects/external/rcn-network/migrations'`

Do you have any insight on this error?

**Comment by rotcivegaf** [comment]
You need create a migrations folder

**Comment by gnidan** [comment]
Huh I thought we stopped requiring a `migrations` directory. If not, maybe we should.

**Comment by rotcivegaf** [comment]
yes, in actual version is required, if you want i can open  an other issue

**Comment by maxsam4** [comment]
I am facing a similar error where I am unable to run migrations/tests on the latest version. The exact error I am getting is 
```
Deploying 'MockOracle'
   ----------------------
Error: Error: Error:  *** Deployment Failed ***

"MockOracle" hit an invalid opcode while deploying. Try:
   * Verifying that your constructor params satisfy all assert conditions.
   * Verifying your constructor code doesn't access an array out of bounds.
   * Adding reason strings to your assert statements.

    at Object.run (/home/max/.nvm/versions/node/v8.12.0/lib/node_modules/truffle/build/webpack:/packages/truffle-migrate/index.js:84:1)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
Truffle v5.0.19 (core: 5.0.19)
Node v8.12.0
```
I am pretty sure that there is something wrong with the compilation or artifacts creation process because I am able to compile contracts with older truffle version (5.0.6) and then run tests/migrations with the latest (5.0.19). Compiling with the latest version and migrating with an older version of truffle gives the same error.

I am not sure on exactly which version it broke. I directly upgraded from 5.0.6 to 5.0.19.

**Comment by eggplantzzz** [comment]
@rotcivegaf Thanks, I found the bug and will put in a PR for it this week! Thanks for the report and repro steps!

**Comment by eggplantzzz** [comment]
PR #2056 addresses this issue.

**Comment by gnidan** [comment]
Sweet! Looks like this will go out in the next release! Thanks all!

**Comment by gnidan** [comment]
The above PR has been released with Truffle v5.0.20. Closing this now! Please let us know if the problem persists.
